eZ/Publish/Core/FieldType/Image/ImageThumbnailStrategy.php

```diff
@@ -43,12 +53,46 @@ public function getFieldTypeIdentifier(): string
 
     public function getThumbnail(Field $field, ?APIVersionInfo $versionInfo = null): ?Thumbnail
     {
-        /** @var \eZ\Publish\SPI\Variation\Values\ImageVariation $variation */
-        $variation = $this->variationHandler->getVariation(
-            $field,
-            $versionInfo ?? new VersionInfo(),
-            $this->variationName
-        );
+        try {
+            /** @var \eZ\Publish\SPI\Variation\Values\ImageVariation $variation */
+            $variation = $this->variationHandler->getVariation(
+                $field,
+                $versionInfo ?? new VersionInfo(),
+                $this->variationName
+            );
+        } catch (SourceImageNotFoundException $e) {
+            $contentDetails = isset($versionInfo->contentInfo) && isset($versionInfo->versionNo)
+                ? sprintf(' Content: %d, Version No: %d', $versionInfo->contentInfo->id, $versionInfo->versionNo)
+                : '';
+
+            $this->logger->warning(
+                sprintf(
+                    'Thumbnail source image generated for %s field and %s variation could not be found (%s).',
+                    $field->fieldTypeIdentifier,
+                    $this->variationName,
+                    $e->getMessage()
+                )
+                . $contentDetails
```

Comment:
If you have already sprintf, concatenation is not necessary, just use sprintf:
```suggestion
                sprintf(
                    'Thumbnail source image generated for %s field and %s variation could not be found (%s).%s',
                    $field->fieldTypeIdentifier,
                    $this->variationName,
                    $e->getMessage(),
                    $contentDetails
                )
```